### ISSUE TITLE
Error when you trying to remove an item from SimpleFormIterator with formDataConsumer 

### ISSUE THREAD

**Issue by Abenezer** [marmelab/react-admin#4962]
**What you were expecting:**

I should be able to remove items (with - button) as i would in normal SimpleFormItterator (without formDataConsumer) 

**What happened instead:**

when i tried to remove items from array (with - Button) i get 
TypeError: Cannot read property 'name' of undefined  (seems the form is still trying to access the removed index  users[1] 

**Steps to reproduce:**
Just copy paste the example inside FormDataConsumer, and press the remove button  

```
<ArrayInput source="users">
    <SimpleFormIterator>
        <TextInput source="name" />

        <FormDataConsumer>
            {({
                formData, // The whole form data
                scopedFormData, // The data for this item of the ArrayInput
                getSource, // A function to get the valid source inside an ArrayInput
                ...rest,
            }) =>
                scopedFormData.name ? (
                    <SelectInput
                        source={getSource('role')} // Will translate to "users[0].role"
                        choices={['admin', 'user']}
                        {...rest}
                    />
                ) : null
            }
        </FormDataConsumer>
    </SimpleFormIterator>
</ArrayInput>
```

**Environment**

* React-admin version: 3.1 
* Last version that did not exhibit the issue (if applicable):
* React version: 16.12
* Browser: Chrome
* Stack trace (in case of a JS error):

TypeError: Cannot read property 'name' of undefined
(anonymous function)

  492 |                <TextInput source="name" />
  493 | 
  494 |                <FormDataConsumer>
> 495 |                    {({
      | ^  496 |                          formData, // The whole form data
  497 |                          scopedFormData, // The data for this item of the ArrayInput
  498 |                          getSource, // A function to get the valid source inside an ArrayInput

**Comment by Abenezer** [comment]
ok, after little investigation i narrowed the problem to  **scopedFormData not defined** after the form array change.  the work around is to use **formData[getSource('field')]**  in place of  scopedFormData.field. everything worked well 

i sill think  scopedFormData was convenient idea. please try it and confirm if this was only a bug in my version.

**Comment by fzaninotto** [comment]
Indeed, that's a documentation bug. The sandbox contains the proper code:

https://github.com/marmelab/react-admin/blob/d5a24d241d3bb19d22f956165dfa99df825026a2/examples/simple/src/posts/PostEdit.js#L80-L87

**Comment by fzaninotto** [comment]
Fixed by #5017
